### ISSUE TITLE
Compiled bindings should support MultiBindings

### ISSUE THREAD

**Issue by davidbritch** [dotnet/maui#5342]
### Description

Given a view model with `Hue`, `Saturation`, and `Luminosity` properties, the following binding expressions can take advantage of compiled bindings (`x:DataType` set on the `ContentPage` to the view model object):

```xaml
<Slider Value="{Binding Hue}" />
```

When the binding expression is changed to `Binding Hue2` a compiler error results. All good so far.

However, the same behaviour doesn't result when you attempt to use compiled bindings in a `MultiBinding`:

```xaml
            <Label>
                <Label.Text>
                    <MultiBinding StringFormat="{}Hue = {0:F2}, Saturation = {1:F2}, Luminosity = {2:F2}">
                        <Binding Path="Hue" />
                        <Binding Path="Saturation" />
                        <Binding Path="Luminosity" />
                    </MultiBinding>
                </Label.Text>
            </Label>
```

While the above code works, changing `Hue` to a property that doesn't exist (e.g. `Hue2`) doesn't result in a compiler error, which suggests that compiled bindings don't support MultiBindings.

Repro case: [DataBindingDemos.zip](https://github.com/dotnet/maui/files/8266361/DataBindingDemos.zip)

### Steps to Reproduce

1. Modify the page in the sample so that the `MultiBinding` binds to a property in the view model that doesn't exist. No compiler error results, when it should.

### Version with bug

Preview 13 (current)

**Comment by XamlTest** [comment]
Verified on Visual Studio Enterprise 2022 main 17.2.0 Preview 2.0 [32223.438.main]. Repro on Windows. Repro Project: [5342.zip](https://github.com/dotnet/maui/files/8279894/5342.zip)

![image](https://user-images.githubusercontent.com/97647558/158719633-de785d3c-cf30-483c-87b6-9035172368b5.png)

**Comment by StephaneDelcroix** [comment]
![it is known](https://user-images.githubusercontent.com/313003/159645629-14f972b5-3687-46ec-9aed-049fb2cf6ffc.gif)

**Comment by tranb3r** [comment]
Is this going to be fixed?

**Comment by StephaneDelcroix** [comment]
#20127 will compile the individual bindings of a CompiledBinding

**Comment by tranb3r** [comment]
@StephaneDelcroix Is this fix going to be released soon?

**Comment by simonrozsival** [comment]
@tranb3r I think that with the correct x:DataType, these bindings should be already compiled in .NET 9 since P7 or RC1 (implemented in https://github.com/dotnet/maui/pull/22023). In principle, you should be able to compile all bindings defined in XAML when you set the right x:DataType.

Could you please try it out with the RC1 release? I won't be able to verify that this has been implemented today, but I will get back to you later if you can't try it out before then.

**Comment by tranb3r** [comment]
@simonrozsival You're absolutely right, it works with net9-rc1, when adding the correct x:DataType.
Thanks!

**Comment by simonrozsival** [comment]
This is related: https://github.com/dotnet/maui/issues/25141

**Comment by simonrozsival** [comment]
This has been implemented in #25152 and merged into the net9.0 branch
